embassy-time/src/queue.rs

```diff
@@ -0,0 +1,617 @@
+//! Generic timer queue implementation
+//!
+//! This module provides a timer queue that works with any executor.
+//!
+//! In terms of performance, this queue will likely be less efficient in comparison to executor-native queues,
+//! like the one provided with e.g. the `embassy-executor` crate.
+//!
+//! # Enabling the queue
+//! - Enable the Cargo feature `generic-queue`. This will automatically instantiate the queue.
+//!
+//! # Initializing the queue
+//! - Call ```unsafe { embassy_time::queue::initialize(); }``` early on in your program, before any of your futures that utilize `embassy-time` are polled.
```

Comment:
IMO we shouldn't require the user to explicitly initialize anything. At every schedule, check if the alarm handle is None and create it if it is. I don't think it'd add much overhead, the current `.unwrap()` is already doing the "check if the alarm handle is None" part at every shedule.

Reply:
I agree and that would solve the "unprotected `Cell`" issue from your first comment.

Only problem is that the creation of the alarm handle then would be protected by the same potentially "disable-all-interrupts" critical-section that protects the inner queue. Have to see if that's possible on the ESP-IDF, as allocating an alarm there calls a bunch of code, that does `malloc` and whatnot. Let me check though.

Reply:
if a particular impl needs initializing stuff at runtime, it can still declare a `fn initialize()` itself and ask the user to call it. I don't think we should require it from `embassy-time` itself if most impls won't need it.